packages/zoe/src/zoeService/zoe.js

```diff
@@ -0,0 +1,388 @@
+        const localIssuerRecord = harden({
+          brand: localBrand,
+          issuer: localIssuer,
+          amountMath: localAmountMath,
+        });
+        issuerTable.registerIssuerRecord(localIssuerRecord);
+        registerIssuerByKeyword(keyword, localIssuerRecord);
+        const localPooledPurse = localIssuer.makeEmptyPurse();
+        brandToPurse.init(localBrand, localPooledPurse);
+
+        /** @type ZoeMint */
+        const zoeMint = harden({
+          getIssuerRecord: () => {
+            return localIssuerRecord;
+          },
+          mintAndEscrow: totalToMint => {
+            const payment = localMint.mintPayment(totalToMint);
+            localPooledPurse.deposit(payment, totalToMint);
+          },
+          withdrawAndBurn: totalToBurn => {
+            const payment = localPooledPurse.withdraw(totalToBurn);
+            localIssuer.burn(payment, totalToBurn);
+          },
+        });
+        return zoeMint;
+      };
+
+      const bundle = installation.getBundle();
+      const addSeatObjPromiseKit = makePromiseKit();
+      const publicFacetPromiseKit = makePromiseKit();
+
+      /** @type {InstanceAdmin} */
+      const instanceAdmin = {
+        addZoeSeatAdmin: async (invitationHandle, zoeSeatAdmin, seatData) => {
+          zoeSeatAdmins.add(zoeSeatAdmin);
+          return E(
+            /** @type Promise<addSeatObj> */ (addSeatObjPromiseKit.promise),
+          ).addSeat(invitationHandle, zoeSeatAdmin, seatData);
+        },
+        hasZoeSeatAdmin: zoeSeatAdmin => zoeSeatAdmins.has(zoeSeatAdmin),
+        removeZoeSeatAdmin: zoeSeatAdmin => zoeSeatAdmins.delete(zoeSeatAdmin),
+        getPublicFacet: () => publicFacetPromiseKit.promise,
+        getTerms: () => instanceRecord.terms,
+        getIssuers: () => instanceRecord.terms.issuers,
+        getBrands: () => instanceRecord.terms.brands,
+        getInstance: () => instance,
+      };
+
+      instanceToInstanceAdmin.init(instance, instanceAdmin);
+
+      /** @type {ZoeInstanceAdmin} */
+      const zoeInstanceAdminForZcf = {
+        makeInvitation: (invitationHandle, description, customProperties) => {
+          const invitationAmount = invitationAmountMath.make(
+            harden([
+              {
+                ...customProperties,
+                description,
+                handle: invitationHandle,
+                instance,
+                installation,
+              },
+            ]),
+          );
+          return invitationMint.mintPayment(invitationAmount);
+        },
+        // checks of keyword done on zcf side
+        saveIssuer: (issuerP, keyword) =>
+          issuerTable.getPromiseForIssuerRecord(issuerP).then(issuerRecord => {
+            registerIssuerByKeyword(keyword, issuerRecord);
+            const { issuer, brand } = issuerRecord;
+            if (!brandToPurse.has(brand)) {
+              brandToPurse.init(brand, E(issuer).makeEmptyPurse());
+            }
+          }),
```

Comment:
This `saveIssuer` method is correct but easy to misread as having a common bug. What's worse is that the type declared for this method
```js
  @property {(issuerP: ERef<Issuer>, keyword: Keyword) => void} saveIssuer
```
if accurate, would mean that the code actually does have the bug. The code actually returns a `Promise<undefined>` (or perhaps we say `Promise<void>`, I'm not sure). This method is remotely called as
```js
        return E(zoeInstanceAdmin)
          .saveIssuer(issuerP, keyword)
          .then(() => {
```
If `saveIssuer` was indeed void return, then the promise for the result of `E{...}.saveIssuer(...)` fulfilling, triggering the `.then` callbacks, would indicate *only* that the `saveIssuer` message had been delivered.

If the body of `saveIssuer` had been enclosed in curlies, the type would have been correct and the `.then` callback would be triggered too soon. The code is correct because the absence of these curlies causes it to return a promise for what it's internal `.then` callback returns. Deepening the illusion, this internal `.then` callback is a curly block with no `return`, which does return `undefined`. 

Terminating this internal block with an unnecessary `return undefined;` and fixing the type makes the correctness of the code easier to see.

```suggestion
        saveIssuer: (issuerP, keyword) =>
          issuerTable.getPromiseForIssuerRecord(issuerP).then(issuerRecord => {
            registerIssuerByKeyword(keyword, issuerRecord);
            const { issuer, brand } = issuerRecord;
            if (!brandToPurse.has(brand)) {
              brandToPurse.init(brand, E(issuer).makeEmptyPurse());
            }
            return undefined;
          }),
```
Also, a comment wouldn't hurt.